test/libsolidity/SyntaxTest.cpp

```diff
@@ -51,13 +51,23 @@ TestCase::TestResult SyntaxTest::run(ostream& _stream, string const& _linePrefix
 	return conclude(_stream, _linePrefix, _formatted);
 }
 
+string SyntaxTest::addPreamble(string const& _sourceCode)
+{
+	// Silence compiler version warning
+	string preamble = "pragma solidity >=0.0;\n";
+	// NOTE: this check is intentionally loose to match weird cases.
+	// We can manually adjust a test case where this causes problem.
+	if (_sourceCode.find("SPDX-License-Identifier:") == string::npos)
```

Comment:
It is possible that the license identifier is not added in corner cases like this

```
contract C {
// SPDX-License-Identifier: GPL-3.0
}
```

Reply:
Yeah, it's not added in this case. I added this as a test case.